connectors/sources/tests/fixtures/sharepoint/fixture.py

```diff
@@ -0,0 +1,248 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Module to handle api calls received from connector."""
```

Comment:
So we're faking the data in Sharepoint cause there's no way to host a Sharepoint Instance locally?

cc @tarekziade

Reply:
yes, this is a compromise to have an e2e test without requiring an external service. Deploying Sharepoint requires a final human intervention on a web page. An improvement in the future will be to script the browser with Cypress or equivalent.

Reply:
If we fake a sharepoint deployment, it's equivalent to unit with mocked api calls. 

Have you considered using a long-running sharepoint deployment (we have one deployment for Workplace Search test), and grab credentials from vaults?